code/iomethods.py

```diff
@@ -5,39 +5,116 @@
 import subprocess
 import sys
 from datetime import timedelta, datetime
-from os import path
 
 import numpy as np
 import numpy.ma as ma
-from netCDF4 import Dataset
+import netCDF4
 
 import utils
 import variables
 
+def _get_fileList_for_binaries(dirPath, startTime, endTime):
+    '''
+        Purpose:: There are two kinds of files in the MERG directory. One has files in binary form, and the other in
+                  netCDF. This function is to get only the files that are in between the startTime and endTime that are
+                  in binary format.
+        Input:: dirPath: Path to where the binary MERG files are.
+                startTime: The cutoff for the earliest files you want
+                endTime: The cutoff for the latest files you want
+        Output:: A list of a path to files that are between startTime and endTime
+        Assumptions:: startTime and endTime are in the format YYYYMMDDHHSS
+                      The filename always ends with -pixel
+                      The date is included in the file name, with the format YYYYMMDDHH
+    '''
+    fileString = os.path.join(dirPath, '*-pixel')
+    fileList = glob.glob(fileString)
+    fileList.sort()
+
+    newFileList = []
+
+    start = datetime.strptime(startTime, '%Y%m%d%H%S')  # Convert start and end times to datetimes to compare with
+    end = datetime.strptime(endTime, '%Y%m%d%H%S')      # the current file's time
+
+    for file in fileList:
+        dateFromFileName = file[5:15]
+        dateAsDateTime = datetime.strptime(dateFromFileName, '%Y%m%d%H')
+
+        if start <= end and start <= dateAsDateTime <= end:
+            newFileList.append(file)
+        elif start <= dateAsDateTime or dateAsDateTime <= end:
+            newFileList.append(file)
+
+    return newFileList
+
+
+def _check_pattern(dirPath, startTime, endTime, tdelta, tRes, flag, filelist):
+    '''
+        Purpose:: refactored pattern recognition to be more modular and usable by other functions that
+                    that wants to obtain the startTime, startFile, endTime, endFile
+    '''
+    
+    startFile = ''
+    endFile = ''
+    currFile = ''
+    hasDelimiter = False
+    startTimeInFile = ''
+    endTimeInFile = ''
+    currTimeInFile = ''
+    tokenCounter = 0
+    filenamePattern = ''
+
+    if flag is True:
```

Comment:
Good start refactoring this function. We can do more here though as the same code is repeated in the if-else statement (expect for the re.split in the for).